homeassistant/components/recorder/__init__.py

```diff
@@ -305,8 +305,82 @@ def _setup_connection(self):
         models.Base.metadata.create_all(self.engine)
         session_factory = sessionmaker(bind=self.engine)
         Session = scoped_session(session_factory)
+        self._check_schema()
         self.db_ready.set()
 
+    def _check_schema(self):
+        """Check if the schema needs to be upgraded."""
+        import homeassistant.components.recorder.models as models
+        schema_changes = models.SchemaChanges
+        current_version = getattr(Session.query(schema_changes).order_by(
```

Comment:
By this point the table has already been created even for current installations, presumably because of `metadata.create_all`. If it's empty we're using the inspection logic as discussed below.